server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/ethereum/erc20/ERC20Wallet.java

```diff
@@ -121,6 +121,10 @@ public BigInteger getTransferGasEstimate(String to, BigDecimal tokens) {
         Transaction tx = Transaction.createEthCallTransaction(credentials.getAddress(), contract.getContractAddress(), FunctionEncoder.encode(function));
         try {
             EthEstimateGas estimateGas = w.ethEstimateGas(tx).send();
+            if (estimateGas.hasError()) {
+                log.error("Error getting gas estimate: {}", estimateGas.getError().getMessage());
+                return null;
```

Comment:
throwing an exception from here didn't log the error message properly. So I log it myself and then let it fail on the null that is returned. (which was happening for ioException already)